src/MySqlConnector/ColumnReaders/ColumnReaderFactory.cs

```diff
@@ -0,0 +1,204 @@
+namespace MySqlConnector.ColumnReaders;
+using MySqlConnector.Protocol;
+using MySqlConnector.Protocol.Payloads;
+using MySqlConnector.Protocol.Serialization;
+
+internal class ColumnReaderFactory
+{
+	internal static IColumnReader GetReader(bool binary, ColumnDefinitionPayload columnDefinition, MySqlConnection connection)
+	{
+		var isUnsigned = (columnDefinition.ColumnFlags & ColumnFlags.Unsigned) != 0;
+		if (binary)
+		{
+			switch (columnDefinition.ColumnType)
+			{
+				case ColumnType.Tiny:
+					if (connection.TreatTinyAsBoolean && columnDefinition.ColumnLength == 1 && !isUnsigned)
+						return BinaryBooleanColumnReader.Instance;
+					return isUnsigned ? BinaryUnsignedInt8ColumnReader.Instance : BinarySignedInt8ColumnReader.Instance;
+
+				case ColumnType.Int24:
+				case ColumnType.Long:
+					return isUnsigned ? BinaryUnsignedInt32ColumnReader.Instance : BinarySignedInt32ColumnReader.Instance;
+
+				case ColumnType.Longlong:
+					return isUnsigned ? BinaryUnsignedInt64ColumnReader.Instance : BinarySignedInt64ColumnReader.Instance;
+
+				case ColumnType.Bit:
+					return BitColumnReader.Instance;
+
+				case ColumnType.String:
+					if (connection.GuidFormat == MySqlGuidFormat.Char36
+					    && columnDefinition.ColumnLength / ProtocolUtility.GetBytesPerCharacter(columnDefinition.CharacterSet) == 36)
+						return Guid36ColumnReader.Instance;
+					if (connection.GuidFormat == MySqlGuidFormat.Char32
+					    && columnDefinition.ColumnLength / ProtocolUtility.GetBytesPerCharacter(columnDefinition.CharacterSet) == 32)
+						return Guid32ColumnReader.Instance;
+					goto case ColumnType.VarString;
+
+				case ColumnType.VarString:
+				case ColumnType.VarChar:
+				case ColumnType.TinyBlob:
+				case ColumnType.Blob:
+				case ColumnType.MediumBlob:
+				case ColumnType.LongBlob:
+				case ColumnType.Enum:
+				case ColumnType.Set:
+					if (columnDefinition.CharacterSet == CharacterSet.Binary)
+					{
+						var guidFormat = connection.GuidFormat;
+						if ((guidFormat is MySqlGuidFormat.Binary16 or MySqlGuidFormat.TimeSwapBinary16
+							    or MySqlGuidFormat.LittleEndianBinary16) && columnDefinition.ColumnLength == 16)
+						{
+							switch (guidFormat)
+							{
+							case MySqlGuidFormat.Binary16:
+								return Guid16ColumnReader.Instance;
+							case MySqlGuidFormat.TimeSwapBinary16:
+								return TimeSwapBinary16ColumnReader.Instance;
+							default:
+								return GuidBytesColumnReader.Instance;
```

Comment:
These classes have somewhat random and inconsistent names. It would make more sense for them to be named consistently with the `MySqlGuidFormat` enum values, e.g., `GuidBinary16ColumnReader`, `GuidTimeSwapBinary16ColumnReader`, `GuidLittleEndianBinary16ColumnReader`.

Will fix in https://github.com/mysql-net/MySqlConnector/pull/1332/commits/e400f960af65cc5e2d604321495cce75e64c22e6.